add support for upto 256 idependantly adjustable cxadc cards, the parameters

AR1972 opened the pull request:
directory has moved to /sys/class/cxadc/cxadc[0-7]/device/parameters fix kernel panic caused cxadc_remove() mucking about with device refrences.

Commit:
add support for 8 idependantly adjustable cxadc cards, the parameters
directory has moved to /sys/class/cxadc/cxadc[0-7]/device/parameters
fix kernel panic caused cxadc_remove() mucking about with device
refrences.

atsampson commented:
The indentation in this is broken - it looks like it's been edited with 4-space soft tabs. Can you run scripts/checkpatch.pl (from the Linux source tree) over it?

harrypm commented:
It would be good to add a sync mode to slave all cards to a single controlled config and if possible an option to synchronous capture on multiple if not all cards available.

AR1972 commented:
> It would be good to add a sync mode to slave all cards to a single controlled config and if possible an option to synchronous capture on multiple if not all cards available.

there are class attributes that if written to could be applied to all devices or a subset of devices, not sure how to do this at the moment but it seems doable.

Commit:
fixup white space

Commit:
add cxacd_suspend() and cxadc_restore() card can now be brought back
into a working state after resume, the hardware has to be re-initialized
and device attributes re-synced, the resume code is largly copy and
pasted from cxadc_probe() so I am reasonably confident that it works.

Commit:
whitespace cleanup to better comply with checkpatch.pl
set max cards to 256
move agc reset into it's own function and call agc_reset() during
suspend to avoid the edge case that a diffirent driver is loaded
after resume.

Commit:
change error message to reflect card limit

Commit:
cxadc_probe() should fail if there is a problem creating sysfs files
or changing ownership.

remove log spam in cxadc_resume().
